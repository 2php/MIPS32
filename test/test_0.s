#Load a number and send it to COM
start:
  .text 0
  la $4, 0x1234
  la $5, 0xBFD003F8 # r5 := COM_ADDR
  sw $4, 0($5) # mem[r5] := r4
  syscall
