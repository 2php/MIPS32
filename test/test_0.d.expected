booting
loading memory from file
80000000 24041234
80000004 3c05bfd0
80000008 34a503f8
8000000c aca40000
80000010 0000000c

fetch instr @ 0x80000000 (10000000000000000000000000000000)
MMU[0x80000000] :  0x00000000
Mem[0x00000000] :  0x24041234 (00100100000001000001001000110100)
001001 | 00000 | 00100 | 0001001000110100
ADDIU    rs: 0   rt: 4   imm:0x1234
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00001234 (00000000000000000001001000110100)
alu_r :  0x00001234 (00000000000000000001001000110100)
R[ 4] <= 0x00001234 (00000000000000000001001000110100)

fetch instr @ 0x80000004 (10000000000000000000000000000100)
MMU[0x80000004] :  0x00000004
Mem[0x00000004] :  0x3c05bfd0 (00111100000001011011111111010000)
001111 | 00000 | 00101 | 1011111111010000
LUI      rs: 0   rt: 5   imm:0xbfd0
R[ 5] <= 0xbfd00000 (10111111110100000000000000000000)

fetch instr @ 0x80000008 (10000000000000000000000000001000)
MMU[0x80000008] :  0x00000008
Mem[0x00000008] :  0x34a503f8 (00110100101001010000001111111000)
001101 | 00101 | 00101 | 0000001111111000
ORI      rs: 5   rt: 5   imm:0x03f8
R[ 5] :  0xbfd00000 (10111111110100000000000000000000)
alu_a <= 0xbfd00000 (10111111110100000000000000000000)
alu_b <= 0x000003f8 (00000000000000000000001111111000)
alu_r :  0xbfd003f8 (10111111110100000000001111111000)
R[ 5] <= 0xbfd003f8 (10111111110100000000001111111000)

fetch instr @ 0x8000000c (10000000000000000000000000001100)
MMU[0x8000000c] :  0x0000000c
Mem[0x0000000c] :  0xaca40000 (10101100101001000000000000000000)
101011 | 00101 | 00100 | 0000000000000000
SW       rs: 5   rt: 4   imm:0x0000
R[ 5] :  0xbfd003f8 (10111111110100000000001111111000)
R[ 4] :  0x00001234 (00000000000000000001001000110100)
alu_a <= 0xbfd003f8 (10111111110100000000001111111000)
alu_b <= 0x00000000 (00000000000000000000000000000000)
alu_r :  0xbfd003f8 (10111111110100000000001111111000)
MMU[0xbfd003f8] :  0x1fd003f8
Mem[0x1fd003f8] <= 0x34 (00110100)

fetch instr @ 0x80000010 (10000000000000000000000000010000)
MMU[0x80000010] :  0x00000010
Mem[0x00000010] :  0x0000000c (00000000000000000000000000001100)
000000 | 00000 | 00000 | 00000 | 00000 | 001100
SPECIAL  rs: 0   rt: 0   rd: 0   sa: 0   SYSCALL
halt
